src/network/networkd-ndisc.c

```diff
@@ -1159,15 +1210,31 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
         if (r < 0)
                 return r;
 
-        r = ndisc_drop_outdated(link, timestamp_usec);
+        r = sd_ndisc_router_get_lifetime(rt, &router_lifetime_sec);
         if (r < 0)
-                return r;
+                return log_link_warning_errno(link, r, "Failed to get lifetime of RA message: %m");
 
-        r = ndisc_start_dhcp6_client(link, rt);
+        r = ndisc_drop_outdated(link, timestamp_usec);
         if (r < 0)
                 return r;
 
-        r = ndisc_router_process_default(link, rt);
+        /* https://datatracker.ietf.org/doc/html/rfc4861
+         * Router Lifetime: A Lifetime of 0 indicates that the router is not a default router
+         * and SHOULD NOT appear on the default router list.
+         */
+        if (router_lifetime_sec == 0) {
+                log_link_debug(link, "Received RA with lifetime = 0, dropping route.");
+
+                r = ndisc_drop_lifetime_zero_router(link, rt);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to remove SLAAC route having lifetime = 0, ignoring: %m");
+        } else {
```

Comment:
When lifetime is zero, it is not necessary to call anything below. So, return earlier.